### PR TITLE
K8SPS-78: operator should throw an error if `volumeSpec` is not specified

### PR DESCRIPTION
**Pull request by pooknull**
https://jira.percona.com/browse/K8SPS-78

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/6c86dd4/config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/6c86dd4/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/6c86dd4/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/6c86dd4/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/6c86dd4/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/6c86dd4/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/6c86dd4/sidecars.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/6c86dd4/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-server-mysql-operator/pull/76/commits/6c86dd4d1f0928f9bea3bcf59f4c7bc9e06216b9
image: `perconalab/percona-server-mysql-operator:PR-76-6c86dd4`

**Review by tplavcic — DISMISSED** [review]

File: api/v1alpha1/perconaservermysql_types.go [hunk 1]
```diff
@@ -330,18 +333,23 @@ func (cr *PerconaServerMySQL) CheckNSetDefaults(serverVersion *platform.ServerVe
...
+		return nil, errors.New("volumeSpec and it's internals should be specified")
+	}
+	if v.PersistentVolumeClaim == nil {
+		return nil, errors.New("pvc should be specified")
+	} else {
+		_, limits := v.PersistentVolumeClaim.Resources.Limits[corev1.ResourceStorage]
+		_, requests := v.PersistentVolumeClaim.Resources.Requests[corev1.ResourceStorage]
+		if !(limits || requests) {
+			return nil, errors.New("pvc's resources.limits[storage] or resources.requests[storage] should be specified")
+		}
```
Comment (egegunes) [inline api/v1alpha1/perconaservermysql_types.go:347]: I don't think we need this`else` here, the function will return if `v.PersistentVolumeClaim == nil`

**Review by egegunes — CHANGES_REQUESTED** [review]

File: api/v1alpha1/perconaservermysql_types.go (hunk 1 above)
Comment (pooknull) [reply api/v1alpha1/perconaservermysql_types.go]: fixed https://github.com/percona/percona-server-mysql-operator/pull/76/commits/47a57c3a3f6f9ec616ae1f4bc15f95d0952304ec

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/47a57c3/config.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/47a57c3/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/47a57c3/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/47a57c3/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/47a57c3/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/47a57c3/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/47a57c3/sidecars.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-76/47a57c3/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-server-mysql-operator/pull/76/commits/47a57c3a3f6f9ec616ae1f4bc15f95d0952304ec
image: `perconalab/percona-server-mysql-operator:PR-76-47a57c3`

**Review by egegunes — APPROVED** [review]

**Pull request merged**